source/test_cvrace_stop.cpp

```diff
@@ -0,0 +1,73 @@
+#include "jthread.hpp"
+#include "condition_variable_any2.hpp"
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <chrono>
+#include <mutex>
+#include <atomic>
+#include <thread>
+#include <cassert>
+#include <cstring>
+using namespace::std::literals;
+
+std::condition_variable_any2* cv = nullptr;
+std::mutex m;
+std::stop_source ss;
+std::stop_token st=ss.get_token();
+
+bool done = false;
+bool waiting=false;
+
+std::condition_variable deleter_cv;
+
+void deleter_thread() {
+    std::unique_lock<std::mutex> lk(m);
+    deleter_cv.wait(lk,[]{return waiting;});
+    done = true;
+    cv->notify_all();
```

Comment:
Is this a common pattern that exists in practice?
ie. calling `cv->notify_all()` and immediately destroying the cv?

Most use-cases like this I've seen have the caller of `wait_until()` responsible for destroying the cv after `wait_until()` returns rather than having the notifying thread responsible for destroying the cv.

It seems generally dangerous to be deleting something that you know another thread still has a reference to.

Reply:
It is a pattern that is required to be supported. It comes from POSIX; see the man page for `pthread_cond_destroy` for an example that does exactly this, and the notes in the description of the destructors for `std::condition_variable` and `std::condition_variable_any`.

Reply:
See also the comments in `test_cvrace_hh.cpp`: https://github.com/josuttis/jthread/blob/9c677af5e57abfcdb6e3be477d6895e5cf881476/source/test_cvrace_hh.cpp#L17

Reply:
Thanks for the clarification.